lib/net.js

```diff
@@ -95,16 +95,13 @@ function noop() {}
 function createHandle(fd, is_server) {
   validateInt32(fd, 'fd', 0);
   const type = TTYWrap.guessHandleType(fd);
+  const server = is_server ? PipeConstants.SERVER : PipeConstants.SOCKET;
   if (type === 'PIPE') {
-    return new Pipe(
-      is_server ? PipeConstants.SERVER : PipeConstants.SOCKET
-    );
+    return new Pipe(server);
   }
 
   if (type === 'TCP') {
-    return new TCP(
-      is_server ? TCPConstants.SERVER : TCPConstants.SOCKET
```

Comment:
This happens to work because `TCPConstants.SERVER === PipeConstants.SERVER` and `TCPConstants.SOCKET === PipeConstants.SOCKET`, but that’s not really something we can guarantee.

For example, if we were to reorder `SocketType` in `src/pipe_wrap.h` so that `IPC` comes first, this would break, and it would be hard to detect why.

So, if we want to go with something like this, we should probably merge these on the C++ side first, so that it’s guaranteed that they have the same values?